### PR TITLE
Implemented Grimlock, Dinobot Leader // Grimlock, Ferocious King & Balan, Wandering Knight

### PR DESCRIPTION
**Pull request by Eleros**
- added Grimlock, Dinobot Leader
- added Autobot and Dinosaur Subtypes
- added TransformedPredicate (to filter for transformed permanents)

- added Balan, Wandering Knight (forgot to switch branch)
- added EquippedMultipleSourceCondition (to check for 2 or more Equips)

### CODE REVIEW

**Comment by theelk801** [comment]
Should we be including silver-bordered cards at all? It seems like it could cause problems.

**Comment by Eleros** [comment]
The Sword was included so I assumed it was alright.
This one also doesn't really affect core besides adding the Autobot creature type which doesn't seem like a big deal?

**Comment by theelk801** [comment]
I guess it's more a question of principle, given that silver-bordered cards aren't "real" cards. I'm not in charge though so it's not up to me.

File: Mage/src/main/java/mage/constants/SubType.java [hunk 1]
```diff
@@ -56,6 +56,7 @@
     ATOG("Atog", SubTypeSet.CreatureType, false),
     ATAT("AT-AT", SubTypeSet.CreatureType, true),
     AUROCHS("Aurochs", SubTypeSet.CreatureType, false),
+    AUTOBOT("Autobot", SubTypeSet.CreatureType, false),
```
Comment (JRHerlehy) [inline Mage/src/main/java/mage/constants/SubType.java]: Autobot is a Fantasy Type.  Please be sure to mark it as such.

File: Mage/src/main/java/mage/constants/SubType.java [hunk 2]
```diff
@@ -99,6 +100,7 @@
     DEMON("Demon", SubTypeSet.CreatureType, false),
     DESERTER("Deserter", SubTypeSet.CreatureType, false),
     DEVIL("Devil", SubTypeSet.CreatureType, false),
+    DINOSAUR("Dinosaur", SubTypeSet.CreatureType, false),
```
Comment (JRHerlehy) [inline Mage/src/main/java/mage/constants/SubType.java]: Until the release of Ixilan, Dinosaur is a Fantasy Type.  Please be sure to mark it as such.

File: Mage.Sets/src/mage/cards/g/GrimlockFerociousKing.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+/**
+ *
+ * @author Saga
+ */
+public class GrimlockFerociousKing extends CardImpl{
+    
+   public GrimlockFerociousKing(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId,setInfo,new CardType[]{CardType.ARTIFACT,CardType.CREATURE},"");
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add("Dinosaur");
```
Comment (JRHerlehy) [inline Mage.Sets/src/mage/cards/g/GrimlockFerociousKing.java]: Please use the new method of adding one or more subtypes to a card.  String methods will be deprecated.

File: Mage.Sets/src/mage/cards/g/GrimlockDinobotLeader.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    static {
+        filter2.add(Predicates.not(new SubtypePredicate(SubType.DINOSAUR))); 
+        filter2.add(Predicates.not(new SubtypePredicate(SubType.VEHICLE))); 
+        filter2.add(Predicates.or(new AbilityPredicate(TransformAbility.class), new TransformedPredicate()));
+    }
+    
+    public GrimlockDinobotLeader(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT,CardType.CREATURE}, "{1}{R}{G}{W}");
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add("Autobot");
```
Comment (JRHerlehy) [inline Mage.Sets/src/mage/cards/g/GrimlockDinobotLeader.java]: Please use the new method of adding one or more subtypes to a card.  String methods will be deprecated.

File: Mage.Sets/src/mage/cards/b/BalanWanderingKnight.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+ */
+public class BalanWanderingKnight extends CardImpl {
+    
+    private static final String rule = "{this} has double strike as long as two or more Equipment are attached to it.";
+
+    public BalanWanderingKnight(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId,setInfo,new CardType[]{CardType.CREATURE},"{2}{W}{W}");
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add("Cat");
+        this.subtype.add("Knight");
```
Comment (JRHerlehy) [inline Mage.Sets/src/mage/cards/b/BalanWanderingKnight.java]: Please use the new method of adding one or more subtypes to a card.

ie, this.subtype.add(SubType.Cat, SubType.Knight);

String methods will be deprecated.

File: Mage.Sets/src/mage/cards/g/GrimlockFerociousKing.java (hunk 3 above)
Comment (Eleros) [inline Mage.Sets/src/mage/cards/g/GrimlockFerociousKing.java]: Which subtypes are you talking about here? card only has one

File: Mage/src/main/java/mage/constants/SubType.java (hunk 1 above)
Comment (Eleros) [inline Mage/src/main/java/mage/constants/SubType.java]: Will do, just set it as false by reflex =x

File: Mage.Sets/src/mage/cards/b/BalanWanderingKnight.java (hunk 5 above)
Comment (JRHerlehy) [inline Mage.Sets/src/mage/cards/b/BalanWanderingKnight.java]: No worries, I haven't pushed the String deprication changes yet.

Comment (Eleros) [inline Mage.Sets/src/mage/cards/b/BalanWanderingKnight.java]: Added the requested corrections if I didn't overlook something..

**Comment by jeffwadsworth** [comment]
I suppose LevelX2 can make a determination.  Personally, I am not a fan of unofficial stuff.

**Comment by Eleros** [comment]
Technically official since it's still made by WotC but I totally get where you and theelk801 are coming from.
Personally, I don't really see the harm in having these cards in the system since people will still be able to choose whether they wanna play with them or not and if some can have fun building Transform/Werewolf-Tribal in EDH or whatever, why not? Not like it's gonna overrun Standard or Modern or whatever. The sword and Grimlock in particular are not even that far off from "real" magic cards just going by abilities, so while I don't think adding all Uncards is a good idea (or feasible even), these in particular seemed fine to me. And it's not like you can't just take them out later again if they do prove to cause issues, right?

That said, I'd totally understand if the request were to be denied. Mostly implemented these cards because I myself and some other people I know would love to play with them.

**Comment by JRHerlehy** [comment]
Well, LevelX2 merged the last one for the Sword.  I see not issues then.

I think the big pushback is mainly that we need to change core stuff in the engine for the cards because they do something slightly odd to justify the "un-ness" of the card.

**Comment by theelk801** [comment]
Yeah, all I really take issue with is the addition of Gold as a color. It's not a color in the game rules, so it shouldn't be a color in the game engine. Instead, maybe have the token just be all colors?

**Pull request merged**